fs-index/benches/resource_index_benchmark.rs

```diff
@@ -84,6 +84,55 @@ fn resource_index_benchmark(c: &mut Criterion) {
         });
     });
 
+    // Benchmark `ResourceIndex::update_one()`
+
+    // First, create a new temp directory specifically for the update_one
+    // benchmark since we will be creating new files, removing files, and
+    // modifying files
+
+    let update_one_benchmarks_dir =
+        TempDir::with_prefix("ark-fs-index-benchmarks-update-one").unwrap();
+    let update_one_benchmarks_dir = update_one_benchmarks_dir.path();
+
+    group.bench_function("index_update_one", |b| {
+        b.iter(|| {
+            // Clear the directory
+            std::fs::remove_dir_all(&update_one_benchmarks_dir).unwrap();
+            std::fs::create_dir(&update_one_benchmarks_dir).unwrap();
+
+            // Create 5000 new files
+            for i in 0..5000 {
+                let new_file =
+                    update_one_benchmarks_dir.join(format!("file_{}.txt", i));
+                std::fs::File::create(&new_file).unwrap();
+                std::fs::write(&new_file, format!("Hello, World! {}", i))
+                    .unwrap();
+            }
+            let mut index: ResourceIndex<Crc32> =
+                ResourceIndex::build(black_box(&update_one_benchmarks_dir))
+                    .unwrap();
+
+            // Create a new file
+            let new_file = update_one_benchmarks_dir.join("new_file.txt");
+            std::fs::File::create(&new_file).unwrap();
+            std::fs::write(&new_file, "Hello, World from new file!").unwrap();
+
+            // Modify an existing file
+            let modified_file = update_one_benchmarks_dir.join("file_0.txt");
+            std::fs::write(&modified_file, "Hello, World from modified file!")
+                .unwrap();
+
+            // Remove an existing file
+            let removed_file = update_one_benchmarks_dir.join("file_1.txt");
+            std::fs::remove_file(&removed_file).unwrap();
+
+            // Update the index
+            let _update_result = index.update_one("new_file.txt").unwrap();
+            let _update_result = index.update_one("file_0.txt").unwrap();
+            let _update_result = index.update_one("file_1.txt").unwrap();
```

Comment:
Right idea of the benchmark, but can we make more calls to `fn update_one`? Right now, there is too much of initialization each iteration, but the measured call takes too few. Maybe we should make the number of "new" or "modified" files to be comparable with number of "existing" files?

Reply:
Agreed. Thanks for the suggestions. I’ve increased the number of `update_one()` calls to handle:
- 1000 file creations
- 1000 file modifications
- 1000 file removals